### PR TITLE
Add sql +, -, *, / symbol arith

### PR DESCRIPTION
**Pull request by xueyumusic**
This PR adds sql +, -, *, / symbol arith based on recent new sql grammar support using calcite. Currently it look only support pql-like ADD, SUB, MULT, DIV function, it would be good to also support common used arith symbol.

### CODE REVIEW

**Comment by sunny209868** [comment]
> This PR adds sql +, -, *, / symbol arith based on recent new sql grammar support using calcite. Currently it look only support pql-like ADD, SUB, MULT, DIV function, it would be good to also support common used arith symbol.

Does this support addition, subtraction, multiplication and division of aggregate indices? like SUM(a) / SUM((b). @xueyumusic

**Comment by xueyumusic** [comment]
Hi, @sunny19930321 . It looks this does not support `SUM(a) / SUM((b)` query. It maybe need more execute modification, thanks

**Comment by sunny19930321** [comment]
> It maybe need more execute modification

I understand.

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java [hunk 1]
```diff
@@ -49,6 +49,17 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CalciteSqlParser.class);
 
+  private static final Map<String, String> FUNC_NAME_MAP;
+
+  // Calcite Sqlkind name to pinot operator name map
+  static {
+    FUNC_NAME_MAP = new HashMap<String, String>();
```
Comment (fx19880617) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java]: Could you rename this to `CALCITE_TO_PINOT_FUNC_NAME_MAP`. Also it's better to make this an immutable map.

**Comment by fx19880617** [comment]
@kishoreg do we support those arithmetic computations in query processing?

File: pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java [hunk 2]
```diff
@@ -422,4 +422,36 @@ public void testSelectionTransformFunction() {
     Assert.assertEquals(
         pinotQuery.getFilterExpression().getFunctionCall().getOperands().get(1).getLiteral().getStringValue(), "v1");
   }
+
+  @Test
+  public void testSqlBaseArith() {
```
Comment (mayankshriv) [inline pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java]: Could you also add tests for nested expressions, for example: `(a * ((b  - c) / d) + e)`

Comment (xueyumusic) [reply pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java]: Thanks for review, @fx19880617 , @kishoreg , @mayankshriv , add a nested expression test case.

**Comment by kishoreg** [comment]
@xueyumusic can we merge this?

**Comment by fx19880617** [comment]
> @xueyumusic can we merge this?

this is already implemented in https://github.com/apache/incubator-pinot/pull/5018

**Pull request closed without merging**